### ISSUE TITLE
Ubuntu - Cant Link Library after Compile

### ISSUE THREAD

**Issue by luisforte** [AyaseCore/mod-LuaEngine#1]
I get prepare the cmake with this módule on azerothcore.

I cloned into Azeroth modules, installed the LuaEngine submodule but cant prepare the build
this line i dont understand 
if( ELUNA )

If i missing something, 

thanks!

**Comment by Ramys** [comment]
Emo

**Comment by t880216t** [comment]
doe's it work? i hav't also.

**Comment by Ramys** [comment]
This.. 
https://github.com/AyaseCore/azerothcore-wotlk/blob/ElunaAzerothWotlk/conf/config.cmake.dist

**Comment by luisforte** [comment]
Even with config ELUNA set the erros comes...

https://i.imgur.com/R52maG9.png

**Comment by luisforte** [comment]
Now with autor instructions, i get 86% build... and error agan...

https://i.imgur.com/DxzPyZL.png

**Comment by luisforte** [comment]
![image](https://user-images.githubusercontent.com/20118021/39511991-bc597ce8-4dc5-11e8-963a-44c4f49fc77f.png)

**Comment by t880216t** [comment]
git 小白，求合并分支的操作命令

**Comment by AyaseCore** [comment]
did you merged my branch? @luisforte

**Comment by AyaseCore** [comment]
如果你是用GitExtensions 可以到你的项目中打开
档案库->管理远程档案库
![image](https://user-images.githubusercontent.com/5546244/39521094-adf36efa-4e3f-11e8-8d40-fd050a31ff35.png)

类似这样加入然后关闭 
命令->合并分支 选择我这条分支 然后合并即可

**Comment by luisforte** [comment]
Hi

i dont have GUi to use you steps, i make all on ubuntu terminal.

Buy yes used your branch, followed all u stepes manually...

And now only one error on ASSERT (asserttrack "something")

I will try more,

**Comment by Ramys** [comment]
@luisforte use a Base da Ayase https://github.com/AyaseCore/azerothcore-wotlk/tree/ElunaAzerothWotlk
e compile por ela.
é a mesma da azeroth porém com as mudanças para o Eluna.

**Comment by luisforte** [comment]
Sim ja fiz isso =/ agora da erro em apenas 1 função, aquele ASSERT da ultima print...

**Comment by Ramys** [comment]
Vou te mandar a minha source que foi perfeitamente ,ok.

**Comment by Ramys** [comment]
This Link Man. https://github.com/AyaseCore/Eluna/commit/89c569ebdfd79c61f759c8630a349d375de059bb
fix a warning.

**Comment by luisforte** [comment]
Now i can compile all project, but when compiler finish and cmake do post job i get this erros 

![image](https://user-images.githubusercontent.com/20118021/39552332-22cbf7ca-4e3f-11e8-937b-d910d4518a37.png)

**Comment by luisforte** [comment]
The "start" os erros

![image](https://user-images.githubusercontent.com/20118021/39553783-b02e5652-4e45-11e8-8a2a-547676f4818f.png)

**Comment by AyaseCore** [comment]
I don't know much about systems other than Windows.
I don't even know how it generates a project.
Because I've only tested it in Windows. It works.
Can it be successfully generated now?

**Comment by luisforte** [comment]
yes generate and compile all fine.

But on finsihed compiling i get this ammount os erros of undefinied reference

**Comment by AyaseCore** [comment]
或者你试试删掉这部分看看 @t880216t

**Comment by luisforte** [comment]
Same error

Even ua file fix

**Comment by AyaseCore** [comment]
32 or 64 bits no distinction.Lua source code already contains.

**Comment by luisforte** [comment]
Yes i understand.

I think the CMAKE are using lua library from my DISTRO insted of Eluna libs

**Comment by AyaseCore** [comment]
i don't know Ubuntu how to Link Library after Compile.
So now I'm going to put the luaEngine files into gamelib, so that I can include the files of eluna when I finally link to lib.
@luisforte

**Comment by luisforte** [comment]
@AyaseCore 
Hi!

The new fix trow this
![image](https://user-images.githubusercontent.com/20118021/39658694-b67dca24-4fee-11e8-89d7-0c7bccf4d260.png)

**Comment by luisforte** [comment]
the new Fix as Succefull Builded.... Thanks @AyaseCore 

Thanks @Ramys

**Comment by AyaseCore** [comment]
Are you using my latest branch of azerothcore?
but ELUNA is disabled by default.
If you enable eluna in cmake, can you build it? @luisforte

**Comment by t880216t** [comment]
@AyaseCore Excuse me , how to make it enable ?

**Comment by luisforte** [comment]
AyaseCore said, enable here 
![image](https://user-images.githubusercontent.com/20118021/39659781-e97396f8-5006-11e8-9465-8303d525b545.png)
I compiled with no error, but i cant test now

**Comment by luisforte** [comment]
![image](https://user-images.githubusercontent.com/20118021/39659789-25910698-5007-11e8-9660-57ea47451185.png)

**Comment by AyaseCore** [comment]
It looks good, but we need to actually test it.

**Comment by luisforte** [comment]
I will do today at 3:00 or less   

Sure i post here the result.... i am VERY anxious  📟

**Comment by luisforte** [comment]
Another "foton" lol

![image](https://user-images.githubusercontent.com/20118021/39659996-fa3c914c-500b-11e8-9bb5-b76cb646112f.png)

**Comment by AyaseCore** [comment]
remove cmake/after_gs_install.cmake
if( WIN32 )
	if ( MSVC )
		add_custom_command(TARGET game
		POST_BUILD
		COMMAND ${CMAKE_COMMAND} -E make_directory "${CMAKE_BINARY_DIR}/bin/$(ConfigurationName)/lua_scripts/extensions/"
		COMMAND ${CMAKE_COMMAND} -E copy_directory "${CMAKE_MOD_ELUNA_ENGINE_DIR}/LuaEngine/extensions" "${CMAKE_BINARY_DIR}/bin/$(ConfigurationName)/lua_scripts/extensions/"
	)
	elseif ( MINGW )
		add_custom_command(TARGET game
		POST_BUILD
		COMMAND ${CMAKE_COMMAND} -E make_directory "${CMAKE_BINARY_DIR}/bin/lua_scripts/extensions/"
		COMMAND ${CMAKE_COMMAND} -E copy_directory "${CMAKE_MOD_ELUNA_ENGINE_DIR}/LuaEngine/extensions" "${CMAKE_BINARY_DIR}/bin/lua_scripts/extensions/"
	)
	endif()
endif()

install(DIRECTORY extensions DESTINATION "${BIN_DIR}/lua_scripts/")
--------

cmake/after_ws_install.cmake
if( WIN32 )
  if ( MSVC )
    add_custom_command(TARGET worldserver
      POST_BUILD
      COMMAND ${CMAKE_COMMAND} -E copy "${CMAKE_MOD_ELUNA_ENGINE_DIR}/conf/mod_LuaEngine.conf.dist" ${CMAKE_BINARY_DIR}/bin/$(ConfigurationName)/
    )
  elseif ( MINGW )
    add_custom_command(TARGET worldserver
      POST_BUILD
      COMMAND ${CMAKE_COMMAND} -E copy "${CMAKE_MOD_ELUNA_ENGINE_DIR}/conf/mod_LuaEngine.conf.dist" ${CMAKE_BINARY_DIR}/bin/
    )
  endif()
endif()

install(FILES "${CMAKE_MOD_ELUNA_ENGINE_DIR}/conf/mod_LuaEngine.conf.dist" DESTINATION ${CONF_DIR})
------

This is a copy of some files that are copied after the build is completed. Temporarily remove it and try to make it successful. If successful, you can then copy that part of the file to your build directory.

**Comment by t880216t** [comment]
i had copy this file form "/azerothcore-wotlk/mod
ules/mod-LuaEngine/LuaEngine/extensions" to "/azerothcore-wotlk/src
/server/game/extensions"
i not sure is it usefull , but as this ,i can install success.
then core start log can find  "[Eluna]: Executed 1 Lua scripts in 0 ms"
![image](https://user-images.githubusercontent.com/16489708/39660076-0abcde94-500e-11e8-8b90-628105eb3845.png)
but there wasnt any message when my player login.
this is my lua_script:
```bash
local PLAYER_EVENT_ON_LOGIN = 3

local function OnLogin(event, player)
    player:SendBroadcastMessage("Hello world")
end

RegisterPlayerEvent(PLAYER_EVENT_ON_LOGIN, OnLogin)
```

**Comment by AyaseCore** [comment]
- -直接发中文就行.你是用什么系统？
你可以试试直接在lua里写print("hello world")
看看加载能不能读取到那个
如果有输出 游戏没反应可能源码hook问题 等我回去再看看

**Comment by t880216t** [comment]
Ubuntu，直接print试过了，没有输入。

**Comment by AyaseCore** [comment]
你的lua脚本放在哪里

**Comment by t880216t** [comment]
bin/lua_scripts/hello.lua

**Comment by luisforte** [comment]
![image](https://user-images.githubusercontent.com/20118021/39665181-609a951a-5066-11e8-8ad0-9ffd4476e3c6.png)

**Comment by t880216t** [comment]
梦神，我这会的脚本能够被执行到了，我啥都没动，哈哈
@AyaseCore  tks sir ,i had do nothing for change , but my scripts can be work success now，

**Comment by AyaseCore** [comment]
闹哪样.我虚拟机都快编译好了你给我说这个.
好吧.因为用虚拟机可以直接测试所以等到今晚提交的应该不会有问题了

I will install the virtual machine to test it. When I fix it, there should be no problem.

**Comment by AyaseCore** [comment]
![image](https://user-images.githubusercontent.com/20118021/39659996-fa3c914c-500b-11e8-9bb5-b76cb646112f.png)
fix in d514a2032953c060440b6ee4415f867774917653 @luisforte, close the issues.
